plugins/outputs/stomp/stomp.go

```diff
@@ -0,0 +1,120 @@
+package stomp
+
+import (
+	ts "crypto/tls"
+	"net"
+
+	"github.com/go-stomp/stomp"
+
+	"github.com/influxdata/telegraf"
+	commontls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+//STOMP ...
+type STOMP struct {
+	Host      string `toml:"host"`
+	Username  string `toml:"username,omitempty"`
+	Password  string `toml:"password,omitempty"`
+	QueueName string `toml:"queueName"`
+	commontls.ClientConfig
+	conn      *ts.Conn
+	netConn   net.Conn
+	stomp     *stomp.Conn
+	serialize serializers.Serializer
+	Log       telegraf.Logger `toml:"-"`
+}
+
+//Connect ...
+func (q *STOMP) Connect() error {
+	tlsConfig, err := q.ClientConfig.TLSConfig()
+
+	if tlsConfig != nil {
+		q.conn, err = ts.Dial("tcp", q.Host, tlsConfig)
+		if err != nil {
+			return err
+		}
+		q.stomp, err = stomp.Connect(q.conn, stomp.ConnOpt.HeartBeat(0, 0), stomp.ConnOpt.Login(q.Username, q.Password))
+		if err != nil {
+			return err
+		}
+	} else {
+		var err error
+		q.netConn, err = net.Dial("tcp", q.Host)
+		if err != nil {
+			return err
+		}
+		q.stomp, err = stomp.Connect(q.netConn, stomp.ConnOpt.HeartBeat(0, 0), stomp.ConnOpt.Login(q.Username, q.Password))
+		if err != nil {
+			return err
+		}
+
+	}
+
+	q.Log.Info("STOMP Connected...")
+	return nil
+}
+
+//SetSerializer ...
+func (q *STOMP) SetSerializer(serializer serializers.Serializer) {
+	q.serialize = serializer
+}
+
+//Write ...
+func (q *STOMP) Write(metrics []telegraf.Metric) error {
+	for _, metric := range metrics {
+		values, err := q.serialize.Serialize(metric)
+		if err != nil {
+			return err
+		}
+		err = q.stomp.Send(q.QueueName, "text/plain",
+			[]byte(values), nil)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
In case there is a single metric that e.g. cannot be serialized, the whole output is stuck as the metrics are kept in the output buffer when `Write` returns an error. This means that (as the metric is kept) the `Write` function will fail until the output buffer will overflow and the malicious metric is overwritten...

I think you should log an error (`q.Log.Error()`) for the failing metric in case serialization fails and continue with the other metrics. This would drop the malicious metrics immediately.

Furthermore, you should use specific error message prefixes to ease finding the source of an error during debugging...
```suggestion
	for _, metric := range metrics {
		values, err := q.serialize.Serialize(metric)
		if err != nil {
			q.Log.Errorf("Serializing metric %v failed: %s", metric, err)
			continue
		}
		err = q.stomp.Send(q.QueueName, "text/plain", []byte(values), nil)
		if err != nil {
			return fmt.Errorf("sending metric failed: %s", err)
		}
	}
```